Sources/AsyncHTTPClient/NIOTransportServices/NWErrorHandler.swift

```diff
@@ -0,0 +1,172 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2020 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(Network)
+
+    import Network
+    import NIO
+    import NIOHTTP1
+    import NIOTransportServices
+
+    /// Enum containing all the DNS errors
+    public enum NWDNSError: Error {
```

Comment:
do these need to be public? If yes, they should probably get an `HTTPClient*` or something prefix. `NWDNSError` is not unlikely to clash in the future

Reply:
After speaking with @lukasa the needs for the DNS error enum is much reduced. Given NIOTS will be handling them. I could possibly remove this completely

Reply:
That'd be even better ofc :)

Reply:
I'll remove it

Reply:
The other two errors, shall i stick them inside HTTPClient?

Reply:
That sounds like a good plan to me!

Reply:
done